backend/src/resolvers.js

```diff
@@ -0,0 +1,67 @@
+const { UserInputError } = require('apollo-server');
+
+module.exports = {
+    Query: {
+      async posts(parent, args, { dataSources }) {
+        return await dataSources.postsApi.getPosts();
+      },
+      async users(parent, args, { dataSources }) {
+        return await dataSources.usersApi.getUsers();
+      }
+    },
+    User: {
+      async posts(parent, args , { dataSources }) {
+        const posts = await dataSources.postsApi.getPosts();
+        return posts.filter(post => post.author === parent.name);
+      }
+    },
+    Post: {
+      author(parent) {
+        return {
```

Comment:
No, this is not how it works. If `User` ever gets more than single attribute, you are screwed here.

Better assign `post.author = user` in your `createPost` with `user` as a complex object. Then you don't have to worry about any additional properties of `user`.